### ISSUE TITLE
Core/Movement: Desync between top MotionMaster and unit movespline

### ISSUE THREAD

**Issue by ccrs** [TrinityCore/TrinityCore#16681]
I've noticed that motion master updates get interrupted on root and stun unitstates, but the unit spline is not. This leads to weird teleport visuals/early movement finish.

I'll try to get some feedback here before openning a PR.

````` diff
void Unit::UpdateSplineMovement(uint32 t_diff)
{
+    // prevent spline to be updated if motionmaster is not
+    if (HasUnitState(UNIT_STATE_ROOT | UNIT_STATE_STUNNED))
+       return;

    if (movespline->Finalized())
        return;

    movespline->updateState(t_diff);
    bool arrived = movespline->Finalized();

    if (arrived)
        DisableSpline();

    m_movesplineTimer.Update(t_diff);
    if (m_movesplineTimer.Passed() || arrived)
        UpdateSplinePosition();
}
````

85c912c4048287931985c98186423d2bf51fdebb

**Comment by Shauren** [comment]
This method is only responsible for updating serverside position - client handles it by itself from the point the spline is started

**Comment by Treeston** [comment]
@Shauren isn't this what causes mobs to teleport if stunned during motion though?

**Comment by ccrs** [comment]
yeah, thats the issue: serverside position is updated while client is not

**Comment by Trisjdc** [comment]
Movespline is supposed to be already finalized if the unit has UNIT_STATE_ROOT or UNIT_STATE_STUNNED. There's a missing MoveSplineInit::Stop call somewhere.

**Comment by ccrs** [comment]
I guess the movementgenerator should check for unit state and send the Stop or Init whenever its necessary.

**Comment by ccrs** [comment]
The problem remains on manual movespline Inits used in other scripts, those wont be handled.

**Comment by Treeston** [comment]
@ccrs any update on this?

**Comment by Rushor** [comment]
relation to https://github.com/TrinityCore/TrinityCore/issues/15952 ?

**Comment by Treeston** [comment]
Likely same bug.

**Comment by Aokromes** [comment]
This bug/fix is still valid after fe68b2e99851a888c5182d7bebddfea67459afd3  ?

**Comment by ccrs** [comment]
that was only a cleanup and renaming, no logic changes

**Comment by Aokromes** [comment]
this issue is still valid?

**Comment by ccrs** [comment]
yes
